dwave_networkx/algorithms/exact_cover.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+from dwave_networkx.utils import binary_quadratic_model_sampler
+from dimod import BinaryQuadraticModel
+
+__all__ = ['is_exact_cover', 'exact_cover']
+
+def is_exact_cover(problem_set, subsets):
+    """Determines whether the given list of subsets is an exact cover of
+    `problem_set`.
+
+    An exact cover is a collection of subsets of `problem_set` that contain every 
+    element in `problem_set` exactly once. More formally, all elements of the 
+    exact cover are disjoint and the union of elements is the original problem_set. 
+
+    Parameters
+    ----------
+    problem_set : iterable
+        An iterable of unique numbers
+
+    subsets : list(iterable(numeric))
+        A list of subsets of `problem_set`  
+
+    Returns
+    -------
+    is_cover : bool
+        True if the given list of subsets forms an exact cover.
+
+    Examples
+    --------
+    This example checks two potential covers for a problem set of 
+    {1, 2, 3, 4, 5}.
+
+    >>> import dwave_networkx as dnx
+    >>> problem_set = {1, 2, 3, 4, 5}
+    >>> subsets = [{1, 2}, {3, 4, 5}]
+    >>> dnx.is_exact_cover(problem_set, subsets)
+    True
+    >>> subsets = [{1, 2, 3}, {3, 4, 5}]
+    >>> dnx.is_exact_cover(problem_set, subsets)
+    False
+
+    """
+    # Verify that problem_set is valid
+    orig_len = len(problem_set)
+    problem_set = set(problem_set)
+    if len(problem_set) != orig_len:
+        raise ValueError("problem_set should not contain any duplicates")
+    
+    # Verify that every element in problem_set is contained exactly once in subsets
+    cover = [x for subset in subsets for x in subset]
+
+    if len(cover) != orig_len:
+        return False
+
+    return all(x in cover for x in problem_set)
```

Comment:
Shorter, and maybe a bit faster:
```suggestion
    return problem_set == set.union(*subsets) and orig_len == sum(map(len, subsets))
```